test/pytest/test_pcp.py

```diff
@@ -0,0 +1,671 @@
+    archive_1.pmiPutValue("kernel.all.load", "5 minute", "5.0")
+    archive_1.pmiPutValue("kernel.all.load", "15 minute", "15.0")
+    archive_1.pmiWrite(0, 0)
+
+    archive_1.pmiEnd()
+
+    return pcp_dir
+
+
+@pytest.fixture
+def instances_change_archive(tmpdir_factory):
+    pcp_dir = tmpdir_factory.mktemp('instances-change-archives')
+    archive_1 = pmi.pmiLogImport(f"{pcp_dir}/0")
+
+    domain = 60  # Linux kernel
+    pmid = archive_1.pmiID(domain, 2, 0)
+    indom = archive_1.pmiInDom(domain, 2)
+    units = archive_1.pmiUnits(0, 0, 0, 0, 0, 0)
+
+    archive_1.pmiAddMetric("network.interface.total.bytes", pmid, PM_TYPE_U32, indom,
+                           PM_SEM_COUNTER, units)
+    archive_1.pmiAddInstance(indom, "lo", 1)
+    archive_1.pmiAddInstance(indom, "eth0", 2)
+    archive_1.pmiAddInstance(indom, "eth1", 3)
+
+    archive_1.pmiPutValue("network.interface.total.bytes", "lo", "1")
+    archive_1.pmiPutValue("network.interface.total.bytes", "eth0", "0")
+    archive_1.pmiPutValue("network.interface.total.bytes", "eth1", "1")
+    archive_1.pmiWrite(1597663539413, 0)
+
+    archive_1.pmiPutValue("network.interface.total.bytes", "lo", "2")
+    archive_1.pmiPutValue("network.interface.total.bytes", "eth0", "1")
+    archive_1.pmiWrite(1597663539483, 0)
+
+    archive_1.pmiEnd()
+
+    return pcp_dir
+
+
+@pytest.fixture
+def instances_rate_archive(tmpdir_factory):
+    pcp_dir = tmpdir_factory.mktemp('instances-rate-archives')
+    archive_1 = pmi.pmiLogImport(f"{pcp_dir}/0")
+
+    domain = 60  # Linux kernel
+    pmid = archive_1.pmiID(domain, 2, 0)
+    indom = archive_1.pmiInDom(domain, 2)
+    units = archive_1.pmiUnits(0, 0, 0, 0, 0, 0)
+
+    archive_1.pmiAddMetric("disk.value", pmid, PM_TYPE_U32, indom,
+                           PM_SEM_INSTANT, units)
+    archive_1.pmiAddInstance(indom, "sda", 1)
+    archive_1.pmiAddInstance(indom, "sdb", 2)
+    archive_1.pmiAddInstance(indom, "sdc", 3)
+
+    archive_1.pmiPutValue("disk.value", "sda", "1")
+    archive_1.pmiPutValue("disk.value", "sdb", "1")
+    archive_1.pmiPutValue("disk.value", "sdc", "1")
+    archive_1.pmiWrite(0, 0)
+
+    archive_1.pmiPutValue("disk.value", "sda", "1")
+    archive_1.pmiPutValue("disk.value", "sdb", "1")
+    archive_1.pmiPutValue("disk.value", "sdc", "2")
+    archive_1.pmiWrite(1, 0)
+
+    archive_1.pmiPutValue("disk.value", "sda", "1")
+    archive_1.pmiPutValue("disk.value", "sdb", "10")
+    archive_1.pmiPutValue("disk.value", "sdc", "3")
+    archive_1.pmiWrite(2, 0)
+
+    archive_1.pmiEnd()
+
+    return pcp_dir
+
+
+@pytest.fixture
+def unsupported_metric_archive(tmpdir_factory):
+    pcp_dir = tmpdir_factory.mktemp('unsupported-metric-archives')
+    archive_1 = pmi.pmiLogImport(f"{pcp_dir}/0")
+    archive_1.pmiAddMetric("mock.value", PM_ID_NULL, PM_TYPE_STRING, PM_INDOM_NULL,
+                           PM_SEM_INSTANT, archive_1.pmiUnits(0, 0, 0, 0, 0, 0))
+
+    archive_1.pmiPutValue("mock.value", None, b"10")
```

Comment:
What exactly makes this unsupported? That binary value instead of string?

Reply:
`PM_TYPE_STRING` is not a metric we support, so it isn't about the value per se but about the type and PM_TYPE_STRING is a b"" in PCP.

Reply:
I see, thanks! A comment would be helpful for our future selves.